### PR TITLE
Add a way to create lazy SSL connections

### PR DESCRIPTION
**Pull request by kipit**
# Description

Add a way to create lazy SSL connection from an array of config.

PhpAmqpLib does not yet provide a way to lazy connect, it would be a good idea to contribute to https://github.com/php-amqplib/php-amqplib/issues/582

### CODE REVIEW

File: src/Driver/DriverFactory.php [hunk 1]
```diff
@@ -87,18 +95,29 @@ protected static function getPeclConnection(array $connection)
...
      */
     protected static function getPhpAmqpLibConnection(array $connection)
     {
-        return new AMQPLazyConnection(
-            $connection['host'],
-            $connection['port'],
-            $connection['user'],
-            $connection['pwd']
-        );
+        if (isset($connection['ssl']) && $connection['ssl']) {
```
Comment (qpautrat) [inline src/Driver/DriverFactory.php:104]: What is the reason of checking `$connection['ssl']` ?

**Review by qpautrat — APPROVED** [review]
👍

File: src/Driver/DriverFactory.php (hunk 1 above)
Comment (kipit) [reply src/Driver/DriverFactory.php:104]: I wanted to keep the previous behaviour : no `ssl` config means no `ssl`
I also wanted to provide a way to config writers to deactivate `ssl` by setting it to false.
You see another way?

File: src/Driver/DriverFactory.php [hunk 2]
```diff
@@ -14,12 +16,18 @@ class DriverFactory
...
-     *     'pwd'  => '<pwdValue>'
+     *     'host'        => '<hostValue>',
+     *     'port'        => '<portValue>',
+     *     'user'        => '<userValue>',
+     *     'pwd'         => '<pwdValue>',
+     *     'ssl'         => true,                   // Optional: default to false
+     *     'ssl_options' => [                       // Optional: default to [], see https://www.php.net/manual/context.ssl.php
+     *         'capath'      => '/etc/ssl/certs',
+     *         'verify_peer' => true,
+     *     ],
```
Comment (fhuitelec) [inline src/Driver/DriverFactory.php:27]: Suggestion vis-à-vis de [la remarque de @qpautrat](https://github.com/Evaneos/evaneos-ng/pull/11379/files#r408025009) :

```suggestion
     *     'ssl'         => [
     *         'enabled'     => false,
     *         'capath'      => '/etc/ssl/certs',
     *         'verify_peer' => true,
     *     ],
```

C'est ultra discutable ceci dit 🤷

File: src/Driver/DriverFactory.php [hunk 3]
```diff
@@ -14,12 +16,18 @@ class DriverFactory
...
      * [
-     *     'host' => '<hostValue>',
-     *     'port' => '<portValue>',
-     *     'user' => '<userValue>',
-     *     'pwd'  => '<pwdValue>'
+     *     'host'        => '<hostValue>',
+     *     'port'        => '<portValue>',
+     *     'user'        => '<userValue>',
+     *     'pwd'         => '<pwdValue>',
+     *     'ssl'         => true,                   // Optional: default to false
```
Comment (fhuitelec) [inline src/Driver/DriverFactory.php:23]: Ça vaudrait le coup de préciser que pour une connexion via l'extension PECL, on ne gère pas le SSL via ces paramètres.

**Review by fhuitelec — APPROVED** [review]
LGTM 🎉

File: src/Driver/DriverFactory.php (hunk 2 above)
Comment (kipit) [reply src/Driver/DriverFactory.php:27]: Je préfère laisser le flag en dehors des options qui sont passées directement au contexte SSL.

File: src/Driver/DriverFactory.php (hunk 3 above)
Comment (kipit) [reply src/Driver/DriverFactory.php:23]: Je l’ai indiqué ici : https://github.com/Evaneos/Burrow/blob/eba7751386020df305b05545aeeb52cff39613dc/src/Driver/DriverFactory.php#L30

Le terme _options_ est surement _misleading_ ici. _SSL connections_, _SSL stuff_ ?

Comment (kipit) [reply src/Driver/DriverFactory.php:23]: Je l’ai indiqué ici : https://github.com/Evaneos/Burrow/blob/eba7751386020df305b05545aeeb52cff39613dc/src/Driver/DriverFactory.php#L30

Le terme _options_ est surement _misleading_ ici. _SSL connections_, _SSL stuff_ ?

File: src/Driver/DriverFactory.php (hunk 1 above)
Comment (qpautrat) [reply src/Driver/DriverFactory.php:104]: >I also wanted to provide a way to config writers to deactivate ssl by setting it to false.

Oh I see. 👍

**Comment by kipit** [comment]
For `amqp-ext` we should follow configuration availables on [the test case](https://github.com/php-amqp/php-amqp/blob/81389c3dd549fc23a0ab85751fb515b770f7369c/tests/amqpconnection_ssl.phpt#L26-L53) used like [this](https://github.com/php-amqp/php-amqp/blob/6d830a999f6c2fd272c77084b6bcbfe7b85c2780/amqp_connection_resource.c#L428-L466) by the underlying layer.

Will be in another PR.

**Pull request merged**